debugger/gdb-extension/gdb_sgx_plugin.py

```diff
@@ -381,12 +382,117 @@ def oe_debugger_cleanup():
+    def __init__(self):
+        self.libs = {}
+        # objdump is used to figure out imports and exports of
+        # a module.
+        self.enabled = shutil.which('objdump') is not None
+        if not self.enabled:
+            self.message('objdump not available. Missing symbols cannot be analyzed.')
+        else:
+            print('oegdb: objdump found. Mising symbols will be analyzed.')
+
+        # Missing symbols in the following modules can be ignored
+        self.ignored_libs = [
+            # dotnet fails to load one of the dependencies of this module.
+            # But that seems to be expected behavior.
+            'libcoreclrtraceptprovider.so'
+        ]
+        # Symbols with the following name can be ignored
+        self.ignored_symbols = [
+            # Many shared libraries are compiled for gprof and take a weak dependency
+            # on this symbol.
+            '__gmon_start__'
+        ]
+
+    def ignore(self, symbol):
+        # Ignore transactional memory functions
+        if symbol.startswith('_ITM_'):
+            return True
+        # ITM new/delete
+        if symbol in ['_ZGTtnam', '_ZGTtdlPv']:
+            return True
+        if symbol in self.ignored_symbols:
+            return True
+
+
+    def message(self, m):
+        # unresolved_color = '\033[1;38;5;170m'
+        unresolved_color = '\033[0;31m'
+        colorize = lambda str, c: ('%s%s\033[0m' % (c, str))
+        print(colorize('oegdb: ' + m, unresolved_color))
+
+
+    def find_missing_symbols(self, libname, lib):
+        for ig in self.ignored_libs:
+            if libname.endswith(ig):
+                self.message('skipping symbol analysis for module %s' % libname)
+                return
+
+        for sym in lib['imports']:
+            if self.ignore(sym):
+                continue
+
+            found = False
+            for l in self.libs.values():
+                if sym in l['exports']:
+                    found = True
+                    break
+            if not found:
+                self.message('unresolved symbol: %s' % sym)
+
+    def process(self, libname):
+        if not self.enabled:
+            return
+
+        print('oegdb: analyzing symbols for module %s' % libname)
+        output = subprocess.run(['objdump', '-T', libname], stdout=subprocess.PIPE)
+
+        imports = set()
+        exports = set()
+        for l in output.stdout.decode('utf-8').split('\n'):
+            und = "*UND*" in l
+            words = l.split()
+            if words:
+                name = words[-1]
+                if und:
+                    imports.add(name)
+                else:
+                    exports.add(name)
+        lib = {
+             name:libname,
+                'imports':imports,
+                'exports':exports
+             }
+        self.libs[libname] = lib
+        self.find_missing_symbols(libname, lib)
+
+missing_symbol_analyzer = None
+
+def oe_debugger_init_hook():
```

Comment:
We don't need the debugger hooks and don't need to create the analyzer by default.